### PR TITLE
Merge absolute imports into feature/abi

### PR DESCRIPTION
**Pull request by michaeldiamant**
Merges https://github.com/algorand/pyteal/pull/283 into `feature/abi` _without_ updating imports for `pyteal.ast.abi` files.

Details:
* The PR temporarily ignores I252 in `pyteal.ast.abi` to keep the diff smaller.   A subsequent PR will remove the temporary ignore and update imports.
* After applying the temporary I252 ignore, several legitimate lint errors remaining.  I fixed them by-hand.
```
flake8 docs examples pyteal scripts tests *.py
pyteal/ir/tealop.py:84:13: I252 Relative imports are banned.
pyteal/ast/__init__.py:141:1: I252 Relative imports are banned.
make: *** [flake8] Error 1
```

### CODE REVIEW

**Comment by michaeldiamant** [comment]
PR is marked as _draft_ because it _includes_ #287.  I'll mark ready for review after merging #287.  The diff will shrink.

File: CONTRIBUTING.md
Comment (tzaffi) [inline CONTRIBUTING.md:10]: 👍 for removing dead link

File: CONTRIBUTING.md
Comment (tzaffi) [inline CONTRIBUTING.md:46]:
```suggestion
A file in this codebase that requires another module/file in this codebase, should import the absolute path of the module/file, not the relative one, using the `from X import Y` method.
```

**Review by tzaffi — APPROVED** [review]
I'm conditionally approving with one suggestion for a non-essential wording improvement.

**Pull request merged**
